src/jit/emitxarch.cpp

```diff
@@ -7852,16 +7852,19 @@ void emitter::emitDispAddrMode(instrDesc* id, bool noDetail)
     else
     {
         // Munge any pointers if we want diff-able disassembly
-        if (emitComp->opts.disDiffable)
+        if (!frameRef && emitComp->opts.disDiffable && (static_cast<size_t>((disp >> 20) + 1) > 1))
```

Comment:
If I'm right to understand the static_cast expression is basically needed to make sure the transformation of disp into "D1FFAB1EH" should only take place when disp is outside of the range (-1M, +1M), it might be worth a code comment.

Reply:
Added comments in code.